### PR TITLE
change zoom factor value to percent

### PR DESCRIPTION
**Pull request by myarjunar**
funded by: Kartoza

### CODE REVIEW

File: src/app/qgsoptions.cpp [hunk 1]
```diff
@@ -1347,7 +1347,7 @@ void QgsOptions::saveOptions()
   mSettings->setValue( QStringLiteral( "/qgis/default_measure_color_green" ), myColor.green() );
   mSettings->setValue( QStringLiteral( "/qgis/default_measure_color_blue" ), myColor.blue() );
 
-  mSettings->setValue( QStringLiteral( "/qgis/zoom_factor" ), spinZoomFactor->value() );
+  mSettings->setValue( QStringLiteral( "/qgis/zoom_factor" ), QgsOptions::getZoomFactorValue( spinZoomFactor ) );
```
Comment (m-kuhn) [inline src/app/qgsoptions.cpp]: No reason to prefix this with `QgsOptions::`

File: src/app/qgsoptions.h [hunk 2]
```diff
@@ -228,6 +228,8 @@ class APP_EXPORT QgsOptions : public QgsOptionsDialogBase, private Ui::QgsOption
     void saveContrastEnhancement( QComboBox *cbox, const QString &name );
     void initMinMaxLimits( QComboBox *cbox, const QString &name, const QString &defaultVal );
     void saveMinMaxLimits( QComboBox *cbox, const QString &name );
+    void setZoomFactorValue( QSpinBox *zoomFactorSpinBox );
+    double getZoomFactorValue( QSpinBox *zoomFactorSpinBox);
```
Comment (m-kuhn) [inline src/app/qgsoptions.h]: We normally don't prefix getters with `get`.

Just call this `zoomFactorValue`

File: src/app/qgsoptions.cpp [hunk 3]
```diff
@@ -2300,3 +2300,25 @@ void QgsOptions::scaleItemChanged( QListWidgetItem *changedScaleItem )
...
+
+double QgsOptions::getZoomFactorValue(QSpinBox *zoomFactorSpinBox)
+{
+  // Get the decimal value for zoom factor. This function is needed because the zoom factor spin box is shown as a percent value.
+  // The minimum zoom factor value is 1.01
+  if ( zoomFactorSpinBox->value() == zoomFactorSpinBox->minimum() ) return 1.01;
+  else return zoomFactorSpinBox->value()/100.0;
+}
+
+void QgsOptions::setZoomFactorValue(QSpinBox *zoomFactorSpinBox)
```
Comment (m-kuhn) [inline src/app/qgsoptions.cpp]: The spacing is wrong here. You can setup a pre-commit hook as described here https://github.com/qgis/QGIS/blob/master/CONTRIBUTE.md so it's done automatically for you.

File: src/app/qgsoptions.cpp [hunk 4]
```diff
@@ -2300,3 +2300,25 @@ void QgsOptions::scaleItemChanged( QListWidgetItem *changedScaleItem )
   addScaleToScaleList( changedScaleItem );
   mListGlobalScales->setCurrentItem( changedScaleItem );
 }
+
+double QgsOptions::getZoomFactorValue(QSpinBox *zoomFactorSpinBox)
+{
+  // Get the decimal value for zoom factor. This function is needed because the zoom factor spin box is shown as a percent value.
+  // The minimum zoom factor value is 1.01
+  if ( zoomFactorSpinBox->value() == zoomFactorSpinBox->minimum() ) return 1.01;
```
Comment (m-kuhn) [inline src/app/qgsoptions.cpp]: Not sure if that's done by astyl automatically meanwhile. We normally break after every if

    if ( xxx )
      return 1.01;

**Review by m-kuhn — CHANGES_REQUESTED** [review]
Hey @myarjunar thanks a lot, I think this will make this setting much easier to understand.

I have left some comments, mostly for the coding style we use for consistency with the rest of the codebase.

File: src/app/qgsoptions.cpp [hunk 5]
```diff
@@ -2300,3 +2300,25 @@ void QgsOptions::scaleItemChanged( QListWidgetItem *changedScaleItem )
   addScaleToScaleList( changedScaleItem );
   mListGlobalScales->setCurrentItem( changedScaleItem );
 }
+
+double QgsOptions::getZoomFactorValue( QSpinBox *zoomFactorSpinBox )
```
Comment (m-kuhn) [inline src/app/qgsoptions.cpp]: I don't think it's required to pass this as a parameter.

This is a class member variable which is available in here anyway. Just use `spinZoomFactor` directly.

**Comment by myarjunar** [comment]
@m-kuhn thanks for your review :+1:  gonna update my PR

**Comment by m-kuhn** [comment]
Nice, codewise it looks good to me now.

Something else that I think could increase UX even a bit more is to add a tooltip to the spinbox, that shortly explains what the number does (something like "Specifies the change in zoom level with each move of the mouse wheel. The bigger the number, the faster zooming with the mouse wheel will be." Feel free to improve on this if you want.

**Comment by nyalldawson** [comment]
Another nice change would be to flip the QSpinBox to a QgsSpinBox, and set the clear value to the default zoom factor. Then you get a handy reset icon embedded in the widget for changing it back to the default value.

**Review by m-kuhn — APPROVED** [review]

**Comment by m-kuhn** [comment]
Thank you @myarjunar 
Excellent work

File: src/app/qgsoptions.cpp [hunk 6]
```diff
@@ -2300,3 +2300,27 @@ void QgsOptions::scaleItemChanged( QListWidgetItem *changedScaleItem )
...
+void QgsOptions::setZoomFactorValue()
+{
+  // Set the percent value for zoom factor spin box. This function is for converting the decimal zoom factor value in the qgis setting to the percent zoom factor value.
+  if ( mSettings->value( QStringLiteral( "/qgis/zoom_factor" ), 2 ) <= 1.01 )
+  {
+    spinZoomFactor->setValue( spinZoomFactor->minimum() );
+  }
+  else
+  {
+    int percentValue = mSettings->value( QStringLiteral( "/qgis/zoom_factor" ), 2 ).toInt() * 100;
```
Comment (myarjunar) [inline src/app/qgsoptions.cpp]: ups I messed up here. it supposed to be `toDouble()`.
gonna create a patch for this.

Comment (m-kuhn) [inline src/app/qgsoptions.cpp]: good catch

**Pull request merged**